aten/src/ATen/TensorIterator.cpp

```diff
@@ -292,16 +292,30 @@ void TensorIteratorBase::reorder_dimensions() {
 // See the [Common Dtype Computation] note
 ScalarType TensorIteratorBase::compute_common_dtype() {
   at::native::ResultTypeState state = {};
+  std::ostringstream input_dtypes;
+
   for (const auto& op : operands_) {
     if (op.is_output) {
       continue;
     }
 
+    // Get input dtypes for error reporting.
```

Comment:
It's OK to improve error messages but this approach is a performance issue.

When will `common_dtype_` be `ScalarType::Undefined`? 

If we are really concerned that will happen and want to test for it regularly then the generation of the error message should only occur AFTER the error state is detected, otherwise we're performing expensive string construction on the default hot path.

cc @ngimel

Reply:
this is a great point, i totally missed it! that said, i just rolled back the changes to errors as per ngimel suggestions (see below)